lib/nmatrix/math.rb

```diff
@@ -1045,4 +1086,53 @@ def dtype_for_floor_or_ceil
       self.__dense_map__ { |l| l.send(op,rhs) }
     end
   end
+
+  # Norm calculation methods
+  # Frobenius norm: the Euclidean norm of the matrix, treated as if it were a vector
+  def fro_norm
+    #float64 has to be used in any case, since nrm2 will not yield correct result for float32
```

Comment:
What is the problem with nrm2? Can you file a bug on this? Maybe you should avoid using nrm2 since it doesn't work for complex types either (#389).